### PR TITLE
Add missing tests and create coverage job on Travis [ECR-3619]

### PR DESCRIPTION
**Pull request by Parfyonator**
Added missing test for 'ExonumClient' and 'Subscriber', made some code refactoring and added code coverage job to '.travis.yml'.

### CODE REVIEW

File: tests/test_exonum_client.py [hunk 1]
```diff
@@ -18,6 +18,16 @@
 
 SYSTEM_ENDPOINT_POSTFIX = "api/system/v1/{}"
 SERVICE_ENDPOINT_POSTFIX = "api/services/{}/{}"
+EXPLORER_ENDPOINT_POSTFIX = "api/explorer/v1/{}"
+
+
+def random_alphanumeric_string(length=32):
+    from uuid import uuid4
+
+    string = str(uuid4())
```
Comment (popzxc) [inline tests/test_exonum_client.py]: I'd suggest to use `random` module with fixed seed, just so the test data will remain the same for every test.

File: tests/test_exonum_client.py [hunk 2]
```diff
@@ -62,14 +87,68 @@ def mock_requests_get(url, params=None):
...
+def mock_requests_post(url, data=None, headers=None):
+    exonum_public_base = EXONUM_URL_BASE.format(EXONUM_PROTO, EXONUM_IP, EXONUM_PUBLIC_PORT)
+    _exonum_private_base = EXONUM_URL_BASE.format(EXONUM_PROTO, EXONUM_IP, EXONUM_PRIVATE_PORT)
+
+    transactions_endpoint = exonum_public_base + EXPLORER_ENDPOINT_POSTFIX.format("transactions")
+    shutdown_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("shutdown")
+    consensus_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("consensus_enabled")
+    peers_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("peers")
+
+    if url == transactions_endpoint:
```
Comment (popzxc) [inline tests/test_exonum_client.py]: Nit: this can be simplified as `if url in [transactions_endpoint, ..]:`.

Comment (Parfyonator) [reply tests/test_exonum_client.py]: I was thinking about that. But what if someone will decite to extend the coverage beyond the 200 response for some endpoints?

Comment (popzxc) [reply tests/test_exonum_client.py]: When this will become the case, we'll think about it ;)

Comment (Parfyonator) [reply tests/test_exonum_client.py]: OK, I'll simplify the mentioned piece of code )

File: tests/test_exonum_client.py (hunk 1 above)
Comment (Parfyonator) [reply tests/test_exonum_client.py]: Should we really fix the seed?

Comment (popzxc) [reply tests/test_exonum_client.py]: Yes, I think it's good. Later we can add fuzz tests with true random, but for unit tests the predictability is a good thing.

**Comment by Parfyonator** [comment]
I guess that this [issue](https://github.com/exonum/exonum-python-client/issues/10) may be closed now.

**Comment by aleksuss** [comment]
Should code coverage checker prevent merge PR if coverage is poor ?

**Comment by aleksuss** [comment]
Also you can add a badge to the README.md with  codecov status
[codecov](https://codecov.io/gh/exonum/exonum-python-client)

**Comment by popzxc** [comment]
>Should code coverage checker prevent merge PR if coverage is poor ?

Not yet.

**Pull request merged**